pkg/edgeview/src/system.go

```diff
@@ -0,0 +1,1022 @@
+	fmt.Printf("     issuer: %s\n", cert.Issuer)
+}
+
+func runConfigItems() {
+	printColor(" - global settings:", colorCYAN)
+
+	// /hostfs is need here to read the memory limit for eve item
+	configitems := getConfigItems()
+
+	configMap := types.NewConfigItemSpecMap()
+	// global
+	for k, g := range configitems.GlobalSettings {
+		//fmt.Printf("key: %s; value %v\n", k, g)
+		s := configMap.GlobalSettings[k]
+		s1 := types.ConfigItemValue{
+			ItemType: s.ItemType,
+			IntValue: s.IntDefault,
+			StrValue: s.StringDefault,
+			BoolValue: s.BoolDefault,
+			TriStateValue: s.TriStateDefault,
+		}
+		if getCfgValue(g) == getCfgValue(s1) {
+			buff := fmt.Sprintf(" %s: %s\n", k, getCfgValue(g))
+			fmt.Printf("%s", buff)
+		} else {
+			buff := fmt.Sprintf(" %s: %s; default %s\n", k, getCfgValue(g), getCfgValue(s1))
+			printColor(buff, colorYELLOW)
+		}
+		//fmt.Printf("   default: %v\n", configMap.GlobalSettings[k])
+	}
+
+	// agent
+	printColor("\n - agent settings:", colorCYAN)
+	for k, g := range configitems.AgentSettings {
+		printColor("  "+k+":  ", colorRED)
+		for k1, g1 := range g {
+			fmt.Printf("    %s, %s\n", k1, getCfgValue(g1))
+		}
+	}
+}
+
+func getConfigItems() types.ConfigItemValueMap {
+	var cfgItem types.ConfigItemValueMap
+	retbytes, err := ioutil.ReadFile("/persist/status/zedagent/ConfigItemValueMap/global.json")
+	if err != nil {
+		return cfgItem
+	}
+	_ = json.Unmarshal(retbytes, &cfgItem)
+	return cfgItem
+}
+
+func getCfgValue(g types.ConfigItemValue) string {
+	value := ""
+	switch g.ItemType {
+	case types.ConfigItemTypeInt:
+		value = strconv.Itoa(int(g.IntValue))
+	case types.ConfigItemTypeBool:
+		value = strconv.FormatBool(g.BoolValue)
+	case types.ConfigItemTypeString:
+		value = g.StrValue
+	case types.ConfigItemTypeTriState:
+		if g.TriStateValue == types.TS_NONE {
+			value = "None"
+		} else if g.TriStateValue == types.TS_DISABLED {
+			value = "Disabled"
+		} else if g.TriStateValue == types.TS_ENABLED {
+			value = "Enabled"
+		} else {
+			value = "un-supported"
+		}
+	default:
+		value = "un-supported"
+	}
+	return value
+}
+
+func getDownload() {
+	pubsubSvs("/run/", "volumemgr", "DownloaderConfig")
+	pubsubSvs("/run/", "downloader", "DownloaderStatus")
+
+	getMetricsMap("/run/downloader/MetricsMap/", nil, true)
+	checkDownload("/persist/downloads")
+	checkDownload("/persist/vault/downloader")
+	checkDownload("/persist/vault/verifier")
+}
+
+func getTop(lines int) {
+	retStr, _ := runCmd("top -n 1 -b", false)
+	if lines == 0 {
+		fmt.Printf("%s\n", retStr)
+	} else {
+		lis := strings.SplitAfter(retStr, "\n")
+		for i, l := range lis {
+			if i < lines {
+				fmt.Printf("%s", l)
+			} else {
+				break
+			}
+		}
+	}
+}
+
+func runPS(opt string) {
+	var item string
+
+	if !rePattern.MatchString(opt) {
+		return
+	}
+	if strings.Contains(opt, "ps/") {
+		opts := strings.SplitN(opt, "ps/", 2)
+		item = opts[1]
+	}
+
+	matches, _ := filepath.Glob("/proc/*/exe")
+	for i, file := range matches {
```

Comment:
Lots of code to screen scrape /proc output. Why not use github.com/shirou/gopsutil/process which is what we use in domainmgr to extract information about the processes?

Reply:
right, github.com/shirou/gopsutil/process seems to get the process related stats, but I need to get more than that, memory, net routes, intf arp, conntrack stats, etc.

Reply:
This function only gets info for the processes.

FWIW there is a netlink package we use to get routes, intf, etc.
I thought Gopi pushed up some fixes to a conntrac package as well a few years back. So we can avoid screen scaping output in many cases.